src/journal/journald-server.c

```diff
@@ -818,6 +818,78 @@ static bool shall_try_append_again(JournalFile *f, int r) {
         }
 }
 
+typedef struct FailedEntry {
+        int r;
+        size_t nb_entries;
+        size_t nb_items;
+        size_t total_size;
+        const char *file;
+        const char *message;
+        RateLimit rl;
+} FailedEntry;
+
+static void flush_failed_entries_logs(FailedEntry* failed_entry, int level) {
+        if (failed_entry->nb_entries == 0)
+                return;
+
+        const char* dest = failed_entry->file ? strjoina(" to file ", failed_entry->file) : "";
+
+        if (failed_entry->nb_entries > 1)
+                log_full_errno(
+                                level,
+                                failed_entry->r,
+                                "Failed to write %zu entries%s (%zu items, %zu bytes)%s: %m",
+                                failed_entry->nb_entries,
+                                dest,
+                                failed_entry->nb_items,
+                                failed_entry->total_size,
+                                failed_entry->message);
+        else
+                log_full_errno(
+                                level,
+                                failed_entry->r,
+                                "Failed to write entry%s (%zu items, %zu bytes)%s: %m",
+                                dest,
+                                failed_entry->nb_items,
+                                failed_entry->total_size,
+                                failed_entry->message);
+
+        failed_entry->nb_entries = 0;
+        failed_entry->nb_items = 0;
+        failed_entry->total_size = 0;
```

Comment:
This function is only about writing the log messages so we should probably move these assignments out of the function and into the macro